java/src/org/openqa/selenium/json/Input.java

```diff
@@ -97,7 +97,18 @@ public String toString() {
       length = MEMORY_SIZE;
     }
 
-    return "Last " + length + " characters read: " + new String(buffer, offset, length);
+    String last = "Last " + length + " characters read: " + new String(buffer, offset, length);
```

Comment:
This fix can be probably a separate PR.

Reply:
Okay, i will remove this from the PR.
It was helpfull to debug the broken session url endpoint.